src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -269,125 +356,30 @@ uct_ud_mlx5_ep_am_zcopy(uct_ep_h tl_ep, uint8_t id, const void *header,
+                                          /* completion */   comp,
+                                          UCT_EP_STAT_AM, UCT_EP_STAT_BYTES_ZCOPY,
+                                          "uct_ud_mlx5_ep_am_zcopy");
 }
 
 static ucs_status_t
 uct_ud_mlx5_ep_put_short(uct_ep_h tl_ep, const void *buffer, unsigned length,
                          uint64_t remote_addr, uct_rkey_t rkey)
 {
-    uct_ud_mlx5_ep_t *ep = ucs_derived_of(tl_ep, uct_ud_mlx5_ep_t);
-    uct_ud_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface,
-                                                uct_ud_mlx5_iface_t);
-    size_t ctrl_av_size = uct_ud_mlx5_ep_ctrl_av_size(ep);
-    struct mlx5_wqe_ctrl_seg *ctrl;
-    struct mlx5_wqe_inl_data_seg *inl;
-    uct_ud_put_hdr_t *put_hdr;
-    uct_ud_neth_t *neth;
-    uct_ud_send_skb_t *skb;
-    size_t wqe_size;
-
-    UCT_CHECK_LENGTH(sizeof(*neth) + sizeof(*put_hdr) + length,
-                     0, iface->super.config.max_inline, "put_short");
-
-    uct_ud_enter(&iface->super);
-
-    skb = uct_ud_ep_get_tx_skb(&iface->super, &ep->super);
-    if (!skb) {
-        uct_ud_leave(&iface->super);
-        return UCS_ERR_NO_RESOURCE;
-    }
-
-    ctrl = iface->tx.wq.curr;
-    /* Set inline segment which has AM id, AM header, and AM payload */
-    inl = uct_ib_mlx5_txwq_wrap_exact(&iface->tx.wq,
-                                      UCS_PTR_BYTE_OFFSET(ctrl, ctrl_av_size));
-    wqe_size = length + sizeof(*put_hdr) + sizeof(*neth);
-    inl->byte_count = htonl(wqe_size | MLX5_INLINE_SEG);
-
-    /* assume that neth and am header fit into one bb */
-    ucs_assert(sizeof(*put_hdr) + sizeof(*neth) < MLX5_SEND_WQE_BB);
-    neth = (void*)(inl + 1);
-    uct_ud_neth_init_data(&ep->super, neth);
-    uct_ud_neth_set_type_put(&ep->super, neth);
-    uct_ud_neth_ack_req(&ep->super, neth);
-
-    put_hdr = (uct_ud_put_hdr_t *)(neth+1);
-    put_hdr->rva = remote_addr;
-
-    uct_ib_mlx5_inline_copy(put_hdr + 1, buffer, length, &iface->tx.wq);
-
-    wqe_size += ctrl_av_size + sizeof(*inl);
-    UCT_UD_EP_HOOK_CALL_TX(&ep->super, neth);
-    uct_ud_mlx5_post_send(iface, ep, 0, ctrl, wqe_size, INT_MAX);
-
-    skb->len = sizeof(*neth) + sizeof(*put_hdr);
-    memcpy(skb->neth, neth, skb->len);
-    uct_ud_iface_complete_tx_inl(&iface->super, &ep->super, skb,
-                                 (char *)skb->neth + skb->len, buffer, length);
-    UCT_TL_EP_STAT_OP(&ep->super.super, PUT, SHORT, length);
-    uct_ud_leave(&iface->super);
-    return UCS_OK;
+    uct_ud_put_hdr_t puth = { .rva = remote_addr };
+    return uct_ud_mlx5_ep_short_common(tl_ep, 0,
+                                       /* inl. header */  &puth, sizeof(puth),
+                                       /* inl. data */    buffer, length,
+                                       /* packet flags */ UCT_UD_PACKET_FLAG_PUT,
+                                       UCT_EP_STAT_PUT, "put_short");
```

Comment:
```
...
UCT_EP_STAT_PUT, "uct_ud_mlx5_ep_put_short");
```
to align with the AM Zcopy

or vice versa

Reply:
done